src/player.cpp

```diff
@@ -10140,7 +10140,9 @@ void player::mend_item( item_location&& obj, bool interactive )
             descr << _( "<color_white>Skills:</color>\n" );
             for( const auto& e : f.first->skills() ) {
                 bool hasSkill = get_skill_level( e.first ) >= e.second;
-                f.second -= !hasSkill;
+				if ( !hasSkill && f.second ) {
```

Comment:
Fixed with the merge, but a number of your diffs were un-indented.  Just something to keep an eye on.